### DISCUSSION TITLE
PBPK model for endogenous compound need help to troubleshoot

### DISCUSSION THREAD

**Discussion by rhishi24niper** [Open-Systems-Pharmacology/Forum#814] [Questions & Problems]
Hi, I am trying to re-create the PBPK model of endogenous compound (G-CDCA), I have followed all the steps given in the attached supplementary information of the article, but I am stuck at final step. Step 3 “simulate” under “Extend PBPK model for G-CDCA in MoBi”, I was unable to simulate and export to xml. I have attached mbp3 file and corresponding supplementary file. Can you help me with what I am missing?
[PBBA.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/4991322/PBBA.zip)

**Comment by PavelBal** [comment]
Can you maybe also provide the link to the article you are refering to?

**Comment by Yuri05** [comment]
@rhishi24niper You did not create the syntesis reaction of `G-CDCA`, thus the model produces const zero results. This is described in the attachment. The reaction must be created in the `existing `reaction building block (which already contains the CYP7A1 metabolization reaction)
![grafik](https://user-images.githubusercontent.com/25061876/88790140-fcd05500-d197-11ea-8040-4946fc3b47ff.png)

**Comment by rhishi24niper** [comment]
> Can you maybe also provide the link to the article you are refering to?

[PBPK-G-CDCA.pdf](https://github.com/Open-Systems-Pharmacology/Forum/files/5005142/PBPK-G-CDCA.pdf)

I have attached an article I am referring.

**Comment by rhishi24niper** [comment]
> @rhishi24niper You did not create the syntesis reaction of `G-CDCA`, thus the model produces const zero results. This is described in the attachment. The reaction must be created in the `existing `reaction building block (which already contains the CYP7A1 metabolization reaction)
> ![grafik](https://user-images.githubusercontent.com/25061876/88790140-fcd05500-d197-11ea-8040-4946fc3b47ff.png)

@Yuri05 Thanks for the help. As suggested I have created a reaction in the existing building block. When I am trying to update simulation from building block it is giving me error message. I have attached the updated MoBi file.
[PBBA-2.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/5005160/PBBA-2.zip)

![image](https://user-images.githubusercontent.com/47068570/89001711-94c95e00-d2c0-11ea-8e46-de14f7495573.png)

**Comment by StephanSchaller** [comment]
I don't have Version 9 yet installed and can't open your project, but it is likely, that you might have sth. wrong in your list of references or you need additional container criteria.
Please provide the message from the Error/Warnings Log (under "Views" you can open the Errors/Warnings)

**Comment by rhishi24niper** [comment]
@StephanSchaller It gives following error massage 

![image](https://user-images.githubusercontent.com/47068570/89002904-e6bfb300-d2c3-11ea-806e-c897c02be79c.png)

**Comment by StephanSchaller** [comment]
@rhishi24niper , this means you are using a parameter "PARAM_G_CDCA..." in the rate formula that is not defined (i.e., not in the list of referenced objects).

To learn how to set up reactions correctly, please see the manual:
https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/setting-up-reaction-network

Hope this helps,
Stephan

**Comment by rhishi24niper** [comment]
@StephanSchaller  That was a great help. I have fixed the rate formula issue. I am having an issue with parameter identification. I am unable to map observed data to corresponding simulation outputs. 

![image](https://user-images.githubusercontent.com/47068570/90718139-ceb1d280-e276-11ea-8d85-65373d76650a.png)

[G-CDCA-S 19Aug.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/5100752/G-CDCA-S.19Aug.zip)

**Comment by StephanSchaller** [comment]
It seems the observed data was imported in a different dimension than what you can simulate (needs to be in molar concentrations)?

**Comment by rhishi24niper** [comment]
@StephanSchaller both are in molar concentrations

![image](https://user-images.githubusercontent.com/47068570/90845898-9ec50680-e32c-11ea-8a92-c31689c680c8.png)

**Comment by msevestre** [comment]
Do you have any observed data used in your simulation? you cannot assign observed data to an output that you never matched before in a simulation.... logically

What do you see when you open the combo box observed data here? any entry?
![image](https://user-images.githubusercontent.com/1041237/90846060-68d85000-e335-11ea-87a2-7bd174de83e9.png)

**Comment by rhishi24niper** [comment]
@msevestre combo box observed data is empty. How do I assign observed data before simulation?

**Comment by rhishi24niper** [comment]
@msevestre  thanks. I added data before simulation and it works. Thanks for help
